packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/editing.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { Form } from 'informed';
+
+import { mergeClasses } from '../../../classify';
+import Button from '../../Button';
+import ShippingRadios from './shippingRadios';
+import defaultClasses from './editing.css';
+
+const Editing = props => {
+    const {
+        handleSubmit,
+        hasShippingMethods,
+        isLoading,
+        selectedShippingMethod,
+        shippingMethods
+    } = props;
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    return (
+        <div className={classes.root}>
+            <h3 className={classes.heading}>Shipping Method</h3>
+            <Form className={classes.form} onSubmit={handleSubmit}>
+                <ShippingRadios
+                    hasShippingMethods={hasShippingMethods}
+                    isLoading={isLoading}
+                    selectedShippingMethod={selectedShippingMethod}
+                    shippingMethods={shippingMethods}
+                />
+                <Button
+                    priority="normal"
+                    type="submit"
+                    disabled={isLoading || !hasShippingMethods}
+                >
+                    {'Continue to Payment Information'}
```

Comment:
Going back to the edit view after submitting data presents a weird UX. The payment info and adjustment components are now displayed on the page but you have a "continue to payment information" button.

[![Image from Gyazo](https://i.gyazo.com/d150063b4b5eace38534e526b63cc921.gif)](https://gyazo.com/d150063b4b5eace38534e526b63cc921)

We may be able to work around this by changing this text if data has already been submitted (meaning we're past the shipping method step). Or we could always open "update" view in a modal.

@jimbo and @soumya-ashok and I had a discussion about this but I'm not sure if we documented it anywhere. I think there should be three states for each component.

1) An initial, unfilled form (currently the edit state)
2) a filled, "read only" view with an edit button
3) A modal with selections (update?)

Edit: After looking at the ticket, [mock #3](https://jira.corp.magento.com/secure/attachment/330380/Venia%20checkout%20-%20Mobile%20-%20Guest%20-%20Edit%20shipping%20method.png) is that "update" view.

Reply:
Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/688480832b90327a5fa9116193c6fd9bf76c0297.